src/main/kotlin/uk/gov/justice/digital/hmpps/deliusapi/service/person/DisabilityService.kt

```diff
@@ -0,0 +1,62 @@
+package uk.gov.justice.digital.hmpps.deliusapi.service.person
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.deliusapi.advice.Auditable
+import uk.gov.justice.digital.hmpps.deliusapi.dto.v1.person.DisabilityDto
+import uk.gov.justice.digital.hmpps.deliusapi.dto.v1.person.NewDisability
+import uk.gov.justice.digital.hmpps.deliusapi.entity.Disability
+import uk.gov.justice.digital.hmpps.deliusapi.entity.Offender
+import uk.gov.justice.digital.hmpps.deliusapi.entity.StandardReference
+import uk.gov.justice.digital.hmpps.deliusapi.exception.BadRequestException
+import uk.gov.justice.digital.hmpps.deliusapi.mapper.DisabilityMapper
+import uk.gov.justice.digital.hmpps.deliusapi.repository.DisabilityRepository
+import uk.gov.justice.digital.hmpps.deliusapi.repository.OffenderRepository
+import uk.gov.justice.digital.hmpps.deliusapi.repository.ReferenceDataMasterRepository
+import uk.gov.justice.digital.hmpps.deliusapi.repository.findByCodeOrThrow
+import uk.gov.justice.digital.hmpps.deliusapi.repository.findByCrnOrNotFound
+import uk.gov.justice.digital.hmpps.deliusapi.service.audit.AuditContext
+import uk.gov.justice.digital.hmpps.deliusapi.service.audit.AuditableInteraction
+import uk.gov.justice.digital.hmpps.deliusapi.service.audit.OffenderAuditContext
+import uk.gov.justice.digital.hmpps.deliusapi.service.referencedata.WellKnownReferenceDataset
+
+@Service
+class DisabilityService(
+  private val offenderRepository: OffenderRepository,
+  private val disabilityRepository: DisabilityRepository,
+  private val referenceDataMasterRepository: ReferenceDataMasterRepository,
+  private val disabilityMapper: DisabilityMapper,
+) {
+
+  @Transactional
+  @Auditable(AuditableInteraction.ADD_DISABILITY)
+  fun createDisability(crn: String, request: NewDisability): DisabilityDto {
+    val offender = offenderRepository.findByCrnOrNotFound(crn)
+
+    val audit = AuditContext.get(AuditableInteraction.ADD_DISABILITY) as OffenderAuditContext
+    audit.offenderId = offender.id
+
+    val disabilityType = referenceDataMasterRepository.findByCodeOrThrow(WellKnownReferenceDataset.DISABILITY_TYPE.code)
+      .standardReferences?.find { it.code == request.type && it.selectable }
+      ?: throw BadRequestException("An invalid Disability Type was provided.")
```

Comment:
Could you swap this for a call to the `ReferenceDataService` ? It was added recently to simplify ref data lookups.

Should end up looking something like this:
```suggestion
    val disabilityType = referenceDataService.getReferenceData(WellKnownReferenceDataset.DISABILITY_TYPE, selectableOnly = true)
```